File: bench/node/primes.js
const N = parseInt(process.env.BENCH_N || "20000", 10);

function isPrime(n) {
  if (n < 2) return false;
  const limit = Math.floor(Math.sqrt(n));   // once — no d*d in the loop
  for (let d = 2; d <= limit; d++) {
    if (n % d === 0) return false;
  }
  return true;
}

let count = 0;
for (let n = 2; n < N; n++) {
  if (isPrime(n)) count++;
}

console.log(count);

File: bench/dotnet/primes.cs
namespace Bench;
static class Primes
{
    static bool IsPrime(int n)
    {
        if (n < 2) return false;
        int limit = (int)Math.Floor(Math.Sqrt(n));
        for (int d = 2; d <= limit; d++)
            if (n % d == 0) return false;
        return true;
    }
    public static void Run(int n)
    {
        int count = 0;
        for (int k = 2; k < n; k++) if (IsPrime(k)) count++;
        Console.WriteLine(count);
    }
}
